### PR TITLE
impl manage following

### PR DESCRIPTION
**Pull request by mmmommm**
ref #108

### CODE REVIEW

**Comment by sivchari** [comment]
Please update push target to dev-1.1. After that, I check your PR.

**Comment by mmmommm** [comment]
@sivchari 
I fixed it, please review.

**Comment by mmmommm** [comment]
Wait until user API example is added to `example_test.go`

File: client.go [hunk 1]
```diff
@@ -44,6 +44,8 @@ type Users interface {
 	RetrieveSingleUserWithUserName(ctx context.Context, userName string, opt ...*RetrieveUserOption) (*UserResponse, error)
 	Followers(ctx context.Context, userID string, opt ...*FollowOption) (*FollowersResponse, error)
 	Following(ctx context.Context, userID string, opt ...*FollowOption) (*FollowingResponse, error)
+	PostFollowing(ctx context.Context, id string, tuid string) (*PostFollowingResponse, error)
+	UndoFollowing(ctx context.Context, suid string, tuid string) (*UndoFollowingResponse, error)
```
Comment (sivchari) [inline client.go:48]: I think some parameters (e.g. tuid, id) are not clearly.
This args are shown user. So, we must define arg name more meaningful.

File: follow.go [hunk 2]
```diff
@@ -121,3 +122,86 @@ func following(ctx context.Context, c *client, userID string, opt ...*FollowOpti
 
 	return &f, nil
 }
+
+// tuid = target_user_id
+func postFollowing(ctx context.Context, c *client, id string, tuid string) (*PostFollowingResponse, error) {
+	if id == "" {
+		return nil, errors.New("post following by id: id parameter is required")
```
Comment (sivchari) [inline follow.go:129]:
```suggestion
		return nil, errors.New("post following by id: id parameter is required")
```

I think ID is not clearly.

**Review by sivchari — CHANGES_REQUESTED** [review]

File: example/user/follow/main.go [hunk 3]
```diff
@@ -19,11 +20,26 @@ func main() {
...
+		panic(err)
+	}
+	fmt.Println(p)
+
+	// undo following
+	d, err := client.UndoFollowing(context.Background(), "source_user_id", "target_user_id")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(d)
```
Comment (sivchari) [inline example/user/follow/main.go:44]: These code must been moved to ExampleXXX.

File: follow.go [hunk 4]
```diff
@@ -121,3 +122,84 @@ func following(ctx context.Context, c *client, userID string, opt ...*FollowOpti
...
+			URL:     req.URL.String(),
+		}
+	}
+
+	return &postFollowing, nil
+}
+
+func undoFollowing(ctx context.Context, c *client, sourceUserID string, targetUserID string) (*UndoFollowingResponse, error) {
+	if sourceUserID == "" {
+		return nil, errors.New("undo following by sourceUserID: sourceUserID parameter is required")
```
Comment (sivchari) [inline follow.go:173]: I think this error message isn't good.
This is because some of the error messages are duplicated.

File: follow_test.go [hunk 5]
```diff
@@ -1454,3 +1454,215 @@ func Test_following(t *testing.T) {
...
+			},
+			want: &gotwtr.PostFollowingResponse{
+				Errors: []*gotwtr.APIResponseError{
+					{
+						Message: "Sorry, that page does not exist, code:34",
+					},
+				},
+			},
+			wantErr: true,
+		},
```
Comment (sivchari) [inline follow_test.go:1560]: Please add more test case.

File: follow_test.go [hunk 6]
```diff
@@ -1454,3 +1454,215 @@ func Test_following(t *testing.T) {
...
+			want: &gotwtr.UndoFollowingResponse{
+				Errors: []*gotwtr.APIResponseError{
+					{
+						Message: "Sorry, that page does not exist, code:34",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
```
Comment (sivchari) [inline follow_test.go:1651]: Same as above.

**Review by sivchari — CHANGES_REQUESTED** [review]

**Review by sivchari — APPROVED** [review]
LGTM

**Pull request merged**